### ISSUE TITLE
registry overrides don't work on upgrade

### ISSUE THREAD

**Issue by lookfirst** [jspm/jspm-cli#882]
This is my override block:

```
    "overrides": {
      "github:siddii/angular-timer@1.3.3": {
        "main": "dist/angular-timer",
        "registry": "jspm",
        "format": "global",
        "dependencies": {
          "angular": "*",
          "moment": "*",
          "humanize-duration": "github:EvanHahn/HumanizeDuration.js@^3.0.0"
        },
        "shim": {
          "dist/angular-timer": {
            "deps": [
              "angular",
              "humanize-duration",
              "moment"
            ]
          }
        }
      },
      "github:EvanHahn/HumanizeDuration.js@3.0.0": {
        "main": "humanize-duration",
        "format": "global"
      }
    }
```

Of course HumanizeDuration releases a new version... 3.1.0. Now my runtime breaks the next time I do a `jspm install`. Beyond waiting for my PR's to get accepted to the registry or hard locking the version of HumanizeDuration, how do I make things more solid?

**Comment by guybedford** [comment]
Thanks, we could do what we do in the jspm registry and have overrides apply to all versions semver compatible with the override version.

**Comment by OrKoN** [comment]
@guybedford could you point out the relevant places in the code of jspm registry and jspm-cli? I would like to contribute to this.

**Comment by guybedford** [comment]
@OrKoN that would be really great if you are interested in that. It would just be at https://github.com/jspm/jspm-cli/blob/master/lib/package.js#L474.

It would just need to loop through `config.pjson.overrides` to match the `package.name` instead of exact name I think, and then ensure that the version is semver compatible with the currently installing version (package.version).

**Comment by guybedford** [comment]
Released in jspm@0.16.0-beta.6.
